src/dsp/rx_fft.cpp

```diff
@@ -105,8 +109,13 @@ void rx_fft_c::get_fft_data(std::complex<float>* fftPoints, unsigned int &fftSiz
         return;
     }
 
+    std::chrono::time_point<std::chrono::steady_clock> now = std::chrono::steady_clock::now();
+    std::chrono::duration<double> diff = now - d_lasttime;
+    d_lasttime = now;
+
     /* perform FFT */
-    do_fft(d_cbuf.linearize(), d_cbuf.size());  // FIXME: array_one() and two() may be faster
+    d_cbuf.erase_begin(std::min((unsigned int)(diff.count() * d_quadrate * 1.001), (unsigned int)d_cbuf.size() - d_fftsize));
```

Comment:
@csete With respect to your question about latency, I specifically added a fudge factor of 1.001 here to ensure that latency would not build up even if the input sample clock is running fast or the audio sample clock is running slow. I haven't observed any noticeable latency between audio & panadapter.

Reply:
I'd actually expect this change to improve synchronization. Currently, when a burst of input samples arrives the corresponding demodulated audio goes into a buffer, whereas the panadapter immediately jumps ahead to the last input sample.

Reply:
Correct. When the waterfall problem occurs, it is often accompanied with an audio delay (like half a second or so). With the patch, sync has been perfect for me so far (again: when using a remote device, as this is my primary way of using GQRX; haven't tried it with a local device yet).